Gulpfile.js

```diff
@@ -92,33 +94,182 @@ function buildBabel(exclude) {
-      })
-        .pipe(source("index.js"))
-        .pipe(buffer())
-        .pipe(errorsLogger())
-        .pipe(compilationLogger(/* rollup */ true))
-        .pipe(gulp.dest(path.join(pkg, "lib")));
+    Object.keys(packages).map(pkg => {
+      let babelVersion = require("./packages/babel-core/package.json").version;
+      const { format, dest, name, filename, minify } = packages[pkg];
+      let { version = babelVersion } = packages[pkg];
+
+      // If this build is part of a pull request, include the pull request number in
+      // the version number.
+      if (process.env.CIRCLE_PR_NUMBER) {
+        const prVersion = "+pr." + process.env.CIRCLE_PR_NUMBER;
+        babelVersion += prVersion;
+        version += prVersion;
+      }
+
+      let letMinifyArgs = [];
+      if (minify) {
+        letMinifyArgs = [
+          process.env.CI ? [] : uglify(),
+          rename({ extname: ".min.js" }),
+          gulp.dest(path.join(pkg, dest)),
+        ];
+      }
+      return pump(
+        rollup({
+          rollup: require("rollup"),
+          input: getIndexFromPackage(pkg),
+          output: {
+            format,
+            name,
+          },
+          external: ["fs", "path"],
+          plugins: [
+            {
+              name: "babel-source",
+              load(id) {
+                const matches = id.match(/packages\/(babel-[^/]+)\/src\//);
+                if (matches) {
+                  // check if browser field exists for this file and replace
+                  const packageFolder = path.join(
+                    __dirname,
+                    "packages",
+                    matches[1]
+                  );
+                  const packageJson = require(path.join(
+                    packageFolder,
+                    "package.json"
+                  ));
+
+                  if (
+                    packageJson["browser"] &&
+                    typeof packageJson["browser"] === "object"
+                  ) {
+                    for (let nodeFile in packageJson["browser"]) {
+                      const browserFile = packageJson["browser"][
+                        nodeFile
+                      ].replace(/^(\.\/)?lib\//, "src/");
+                      nodeFile = nodeFile.replace(/^(\.\/)?lib\//, "src/");
+                      if (id.endsWith(nodeFile)) {
+                        if (browserFile === false) {
+                          console.log(`Ignore ${nodeFile}`);
+                          return "";
+                        }
+                        console.log(`Replace ${nodeFile} with ${browserFile}`);
+                        return fs.readFileSync(
+                          path.join(packageFolder, browserFile),
+                          "UTF-8"
+                        );
+                      }
+                    }
+                  }
+                }
+              },
+              resolveId(importee) {
+                const matches = importee.match(/^@babel\/([^/]+)$/);
```

Comment:
I've created https://github.com/Andarist/lerna-alias pretty much for this exact thing - aliasing to respective `src`.

Reply:
But it does not replace lib with src in theain entry?

Reply:
It does exactly this (or I've misunderstood your question) - it searches for packages managed by lerna (using lerna's algo - although it's copied) and creates an alias map from `PKG_NAME` to `PKG_PATH/src`